server/app/api/plugin_management.py

```
@@ -0,0 +1,105 @@
+            # Extract safely to staging
+            staging = Path(tmpdir) / "staging"
+            staging.mkdir(parents=True, exist_ok=True)
+            with zipfile.ZipFile(tmp_path) as zf:
+                for member in zf.infolist():
+                    # prevent path traversal
+                    dest = staging / member.filename
+                    dest_abs = dest.resolve()
+                    if not str(dest_abs).startswith(str(staging.resolve())):
+                        raise HTTPException(
+                            status_code=400, detail="Invalid archive paths"
+                        )
```

Ruff (0.12.2)

Abstract `raise` to an inner function

(TRY301)